ckanext/dataset_subscriptions/actions/whatsapp_notifications.py

```diff
@@ -96,3 +110,77 @@ def _validate_plugin_extras(extras):
     for field in CUSTOM_FIELDS:
         out_dict[field['name']] = extras.get(field['name'], field['default'])
     return out_dict
+
+
+def get_phonenumber(user_dict):
+    if "activity_streams_whatsapp_notifications" in user_dict and "phonenumber" in user_dict:
```

Comment:
this function does too many thing at once. The name suggest it's getting the phonenumber form user dict. But here it also check if user enabled the whatsapp notifications.
The notification enabled check should happen separatly IMO.
Then the phonenumber function would look like:
```
def get_phonemuber(user_dict):
   return user_dict.get('phonenumber') # return None if no key present in the dict
```
This function is so simple that I'd rather inline this code instead of encapsulating it in a function.

Reply:
I agree it's pretty simple, but if we'll ever modify the logic to get the phone number we'll have to worry about changing it (possibly) in more than one place. I'd prefer to keep it as a separate function, it's more readable this way (at lest to me ;) )